src/compiler/frontend/reader.rs

```diff
@@ -31,14 +31,18 @@ pub mod tests {
     use super::Reader;
 
     pub fn parse_datum(inp: &str) -> Datum {
+        parse_datum_all(inp)[0].clone()
+    }
+
+    pub fn parse_datum_all(inp: &str) -> Vec<Datum> {
         let mut registry = Registry::new();
         let source = registry
             .add(&mut BufferSource::new(inp, "datum-parser-test"))
             .unwrap();
         let reader = Reader::new();
         let datum = reader.parse(&source).unwrap();
 
-        datum.to_vec()[0].clone()
+        datum.to_vec().clone()
```

Comment:
No need to clone. Instead add a way to turn the parse result into the vector and move it out